app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/components/NetworkStateBar.kt

```diff
@@ -0,0 +1,308 @@
+        NonAnimatedBar(height = 16.dp, color = Color(0xFF36393F))
+    }
+}
+
+@Composable
+fun NonAnimatedBar(
+    height: Dp,
+    color: Color
+) {
+    Box(
+        modifier = Modifier
+            .width(6.dp)
+            .height(height)
+            .padding(end = 2.dp)
+            .background(color, shape = RoundedCornerShape(3.dp))
+    )
+}
+
+
+sealed class ConnectionState {
+    object Available : ConnectionState()
+    object Unavailable : ConnectionState()
+    object AirplaneModeOn : ConnectionState()
+    object AirplaneModeOff : ConnectionState()
+}
+
+
+/**
+ * Network utility to get current state of internet connection
+ */
+val Context.currentConnectivityState: ConnectionState
+    get() {
+        val connectivityManager =
+            getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        return getCurrentConnectivityState(connectivityManager)
+    }
+
+
+private fun getCurrentConnectivityState(
+    connectivityManager: ConnectivityManager
+): ConnectionState {
+    val connected = connectivityManager.allNetworks.any { network ->
+        connectivityManager.getNetworkCapabilities(network)
+            ?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+            ?: false
+    }
+
+
+    return if (connected) {
+        ConnectionState.Available
+    } else {
+        ConnectionState.Unavailable
+    }
+}
+
+/**
+ * Network Utility to observe availability or unavailability of Internet connection
+ */
+@ExperimentalCoroutinesApi
+fun Context.observeConnectivityAsFlow() = callbackFlow {
+    val connectivityManager = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+    val callback = networkCallback { connectionState -> trySend(connectionState) }
+
+    val networkRequest = NetworkRequest.Builder()
+        .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+        .build()
+
+    connectivityManager.registerNetworkCallback(networkRequest, callback)
+
+    val currentState = getCurrentConnectivityState(connectivityManager)
+    trySend(currentState)
+
+    awaitClose {
+        connectivityManager.unregisterNetworkCallback(callback)
+    }
+}
+
+
+fun networkCallback(callback: (ConnectionState) -> Unit): ConnectivityManager.NetworkCallback {
+    return object : ConnectivityManager.NetworkCallback() {
+        override fun onAvailable(network: Network) {
+            callback(ConnectionState.Available)
+        }
+
+        override fun onLost(network: Network) {
+            callback(ConnectionState.Unavailable)
+        }
+    }
+}
+
+@ExperimentalCoroutinesApi
+@Composable
+fun connectivityState(): State<ConnectionState> {
+    val context = LocalContext.current
+
+    return produceState(initialValue = context.currentConnectivityState) {
+        context.observeConnectivityAsFlow().collect { value = it }
+    }
+}
+
+
+@Composable
+fun isAirplaneMode(): State<ConnectionState> {
+    val context = LocalContext.current
```

Comment:
@yugeshjain Composable methods should not return anything. you can just return the ConnectionState directly.

Reply:
FIxed it